src/main/java/hudson/plugins/sauce_ondemand/SauceOnDemandBuildAction.java

```diff
@@ -151,15 +151,30 @@ public static JenkinsBuildInformation retrieveBuildFromSauce(JenkinsSauceREST sa
         JenkinsBuildInformation buildInformation = new JenkinsBuildInformation(buildNumber);
 
         logger.fine("Performing Sauce REST retrieve results for " + buildNumber);
-        String jsonResponse = sauceREST.getBuild(buildNumber);
-        if ("".equals(jsonResponse)) {
-            logger.log(Level.WARNING, "Sauce REST API get build JSON Response was empty for " + buildNumber);
-            return buildInformation;
-        }
 
-        JSONObject buildObj = new JSONObject(jsonResponse);
+        int retries = 0;
+        int maxRetries = 10;
+        String jsonResponse = "";
+
+        while (retries < maxRetries && "".equals(jsonResponse)) {
+            try {
+                jsonResponse = sauceREST.getBuild(buildNumber);
+                if (!"".equals(jsonResponse)) {
+                    JSONObject buildObj = new JSONObject(jsonResponse);
+                    buildInformation.populateFromJson(buildObj);
+                    break;
+                }
+            } catch (Exception e) {
+                jsonResponse = "";
+            }
+            logger.log(Level.WARNING, "Sauce REST API get build JSON Response was empty or threw an exception for " + buildNumber + ", waiting and retrying");
+            retries++;
+            try {
+                Thread.sleep(10000);
```

Comment:
will this delay users being able to access the results by 10 - 100 seconds? does 10 seconds seem to be the time for the build data to be available or can we shorten the sleep so we don't block people from viewing results right away?

Reply:
Yes, but only if the results aren't available.  It should really only be a few seconds but it seems like our backend is pretty slow so it was taking somewhere between like 20-60 seconds before results would be available.

I think taking longer is better than not having results.